lbrynet/daemon/auth/server.py

```diff
@@ -141,6 +142,22 @@ def _deprecated_wrapper(f):
             return f
         return _deprecated_wrapper
 
+    @staticmethod
+    def requires(*components, **component_conditionals):
+        def _wrap(fn):
+            @wraps(fn)
+            def _inner(*args, **kwargs):
+                if args[0].component_manager.all_components_running(*components):
+                    return fn(*args, **kwargs)
+                if component_conditionals:
+                    for component_name, condition in component_conditionals.iteritems():
+                        assert callable(condition)
+                        if not condition(args[0].component_manager.get_component(component_name)):
+                            raise Exception("not all required component conditions are met")
```

Comment:
This should be an exception type defined in `lbrynet.core.Error`, like `ComponentsNotStarted(" ".join(component_names))`